specification/paths/ImportShipment.json

```diff
@@ -0,0 +1,62 @@
+{
+  "post": {
+    "tags": [
+      "RPC"
+    ],
+    "security": [
+      {
+        "OAuth2": [
+          "shipments.manage"
+        ]
+      }
+    ],
+    "summary": "Import shipment.",
+    "description": "This endpoint is used to import externally registered shipments to provide tracking information.",
+    "requestBody": {
+      "description": "The shipment that should be imported.",
+      "required": true,
+      "content": {
+        "application/vnd.api+json": {
+          "schema": {
+            "type": "object",
+            "required": [
+              "data"
+            ],
+            "properties": {
+              "data": {
+                "required": [
+                  "tracking_code"
+                ],
+                "allOf": [
+                  {
+                    "$ref": "#/components/schemas/Shipment"
+                  },
+                  {
+                    "properties": {
+                      "relationships": {
+                        "required": [
+                          "contract"
+                        ]
+                      }
+                    }
+                  }
+                ],
+                "properties": {
+                  "tracking_code": {
```

Comment:
Can you point me towards an example because removing the readOnly property from the component and then doing something like this
 ```
"tracking_code": {
    "$ref": "#/components/schemas/TrackingCode",
    "readonly": true
},```

Isn't working

Reply:
Yes, merging a $ref with something needs allOf, I have found a merged readonly example:
https://github.com/MyParcelCOM/api-specification/blob/develop/specification/schemas/Manifest.json#L28